java-app/src/main/java/com/solace/samples/cloudfoundry/javaapp/controller/SolaceController.java

```diff
@@ -126,7 +126,37 @@ public void init() {
 		logger.info("Solace client initializing and using Credentials: " + solaceCredentials.toString(2));
 
 		final JCSMPProperties properties = new JCSMPProperties();
-		properties.setProperty(JCSMPProperties.HOST, solaceCredentials.getString("smfHost"));
+
+		
+		// The host property is in a json array. Two hosts are provided in a High Availability environment,
+		// one for the primary router and one for the backup. 
+		JSONArray hostsArray = solaceCredentials.getJSONArray("smfHosts");
+
+		// Make a host list (for HA and non HA)
+		String host = "";
+		for( int i = 0; i < hostsArray.length(); i++) {
+			String newHostEntry = hostsArray.getString(i);
+			if( i > 0 )
+				host += ",";
+			
+			host +=  newHostEntry;
+		}
+
+		logger.info("Using host " + host);
+		properties.setProperty(JCSMPProperties.HOST, host);
+		
+		// Must be using HA to have more than 1 host.
+		if( hostsArray.length() > 1 ) {
+			
+			// A Sample for High Availability automatic reconnects.
+			JCSMPChannelProperties channelProperties = (JCSMPChannelProperties) properties
+		            .getProperty(JCSMPProperties.CLIENT_CHANNEL_PROPERTIES);
+			channelProperties.setConnectRetries(1);
+			channelProperties.setReconnectRetries(5);
+			channelProperties.setReconnectRetryWaitInMillis(3000);
+			channelProperties.setConnectRetriesPerHost(20);
+		}
```

Comment:
Can we push this code into the SolaceMessagingServiceInfo class as a convenience method?